stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Attachment.kt

```diff
@@ -40,9 +40,28 @@ public data class Attachment(
 
 ) : CustomObject {
 
+    /**
+     * Represents various states in attachment upload lifecycle.
+     */
     public sealed class UploadState {
-        public object InProgress : UploadState()
+        /**
+         * Idle state before attachment starts to upload.
+         */
+        public object Idle : UploadState()
+
+        /**
+         * State representing attachment upload progress.
+         */
+        public data class InProgress(val bytesRead: Long, val totalBytes: Long) : UploadState()
```

Comment:
We should also review the param names here, especially `bytesRead` is very off

Reply:
Not sure how I did that? I came up with `bytesUploaded` 🤔